x-pack/plugins/observability_solution/entity_manager/server/lib/auth/api_key/api_key.ts

```diff
@@ -45,17 +45,17 @@ export const checkIfEntityDiscoveryAPIKeyIsValid = async (
 
   server.logger.debug('validating API key has runtime privileges for entity discovery');
 
-  return canRunEntityDiscovery(esClient);
+  return canManageEntityDefinition(esClient);
 };
 
 export const generateEntityDiscoveryAPIKey = async (
   server: EntityManagerServerSetup,
   req: KibanaRequest
 ): Promise<EntityDiscoveryAPIKey | undefined> => {
-  const apiKey = await server.security.authc.apiKeys.grantAsInternalUser(req, {
+  const apiKey = await server.security.authc.apiKeys.create(req, {
```

Comment:
Correct, after looking we should be using grantAsInternalUser. Found this in slack that aligns with our use case. This also means we don't require the user enabling/disabling EEM to have the manage_key permissions since this would be done by the internal user. I'll update and can verify this with the automated tests

> One important point is that grantAsInternalUser is intended for API Keys where the user never gets a copy of the API key itself.
It’s for cases where Kibana needs to use an API Key that represents the user’s access, as part of the implementation of a Kibana feature. That is, the Kibana feature uses API keys under the covers (like alerting), and it’s not intended to be visible to the user.